client/src/components/Donation/PaypalButton.tsx

```diff
@@ -13,30 +14,88 @@ import envData from '../../../../config/env.json';
 import { signInLoadingSelector, userSelector } from '../../redux';
 import PayPalButtonScriptLoader from './PayPalButtonScriptLoader';
 
-const { paypalClientId, deploymentEnv } = envData;
-export class PaypalButton extends Component {
-  constructor(props) {
+type PaypalButtonProps = {
+  addDonation: (data: AddDonationData) => void;
+  donationAmount: number;
+  donationDuration: string;
+  handleProcessing: (
+    duration: string,
+    amount: number,
+    action: string
+  ) => unknown;
+  isDonating: boolean;
+  onDonationStateChange: ({
+    redirecting,
+    processing,
+    success,
+    error
+  }: {
+    redirecting: boolean;
+    processing: boolean;
+    success: boolean;
+    error: string | null;
+  }) => void;
+  skipAddDonation?: boolean;
+  t: (label: string) => string;
+  theme: string;
+  isSubscription?: boolean;
+};
+
+type PaypalButtonState = {
+  amount: number;
+  duration: string;
+  planId: string | null;
+};
+
+export interface AddDonationData {
+  redirecting: boolean;
+  processing: boolean;
+  success: boolean;
+  error: string | null;
+}
+
+const {
+  paypalClientId,
+  deploymentEnv
+}: { paypalClientId: string | null; deploymentEnv: 'staging' | 'live' } =
+  envData as {
+    paypalClientId: string | null;
+    deploymentEnv: 'staging' | 'live';
+  };
+
+export class PaypalButton extends Component<
+  PaypalButtonProps,
+  PaypalButtonState
+> {
+  static displayName = 'PaypalButton';
+  constructor(props: PaypalButtonProps) {
     super(props);
     this.handleApproval = this.handleApproval.bind(this);
   }
 
-  state = {};
+  // I can't figure out if it is necessary
+  // TODO: determine if this needs initializing
```

Comment:
```suggestion
```
@ahmadabdolsaheb if you've checked this works without an explicit initialization then these comments can go.